app/src/main/java/uk/gov/onelogin/MainActivityViewModel.kt

```diff
@@ -39,11 +39,18 @@ class MainActivityViewModel @Inject constructor(
 
     override fun onPause(owner: LifecycleOwner) {
         super.onPause(owner)
-        if (bioPrefHandler.getBioPref() != BiometricPreference.NONE &&
-            tokenRepository.getTokenResponse() != null
-        ) {
+        if (checkBiometrics() != null && tokenRepository.getTokenResponse() != null) {
             tokenRepository.clearTokenResponse()
             navigator.navigate(LoginRoutes.Start)
         }
     }
+
+    private fun checkBiometrics(): BiometricPreference? {
```

Comment:
I'm not sure this method particularly makes things any clearer here. Perhaps something more like this:
```suggestion
    private fun biometricPrefIsNone() = bioPrefHandler.getBioPref() == BiometricPreference.NONE
```